app/models/piece.rb

```diff
@@ -5,6 +5,19 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
+  # Returns "valid" and updates the piece's x and
+  # y position to provided coordinates if move is valid,
+  # otherwise do nothing and return nil.
+  def move(x,y)
+    if valid_move?(x,y)
+      self.x_position = x
+      self.y_position = y
+      return "valid"
```

Comment:
Exactly. With a bang, you could technically infer that if the method doesn't raise an exception and break the program things are valid.

I don't really like that though, because sometimes having the boolean is just plain useful. Typically I'm for better to ask for forgiveness than permission, but sometimes both is good.